Update Gsoc Guide for 2022 (#1017)
mattermost/mattermost-developer-documentation: site/content/contribute/gsoc/_index.md

   * If you have general questions about participating in GSoC with Mattermost, ask in the [GSoC channel](https://community.mattermost.com/core/channels/google-summer-of-code) or send an email to gsoc@mattermost.com
1. Work with the project mentors on your application: they are there to help you write a good proposal!
1. Submit your application before the deadline. All applications must go through Google's official application process.

Make sure to take a look at [the full Google Summer of Code timeline](https://summerofcode.withgoogle.com/how-it-works/#timeline) for more details!

## Mentors
Here are our mentors for this year, the links lead to their profiles on our community server

-   Technical Engineering Mentors:
    -   Devin Binnie ([@devin.binnie](https://community.mattermost.com/core/messages/@devin.binnie))
    -   Daniel Espino García ([@daniel.espino.garcia](https://community.mattermost.com/core/messages/@daniel.espino.garcia))
    -   Alejandro García Montoro ([@alejandro.garcia](https://community.mattermost.com/core/messages/@alejandro.garcia))
    -   Elias Nahum ([@elias.nahum](https://community.mattermost.com/core/messages/@elias.nahum))

>     -    Claudio Costa ([@claudio.costa](https://community.mattermost.com/core/messages/@claudio.costa))

Reviewer: Just one more addition and I think we can merge! :tada:
```suggestion
    -    Claudio Costa ([@claudio.costa](https://community.mattermost.com/core/messages/@claudio.costa))
    -    Ibrahim Acikgoz ([@ibrahim.acikgoz](https://community-daily.mattermost.com/core/messages/@ibrahim.acikgoz))
```

description: "Ideas list for Google Summer of Code"
---

# Google Summer of Code

> This page outlines everything you need to know about contributing to Mattermost for the Google Summer of Code (GSoC).

Reviewer: Do we need the article? Not sure, actually, so this is more a question than a request for changes :)
```suggestion
This page outlines everything you need to know about contributing to Mattermost for Google Summer of Code (GSoC).
```

## Contributor Guidance
1. Read everything in this doc, we've put this together to maximize your chances of drafting a successful GSoC proposal.

> 1. Learn about our [contribution process](https://developers.mattermost.com/contribute/getting-started/). We encourage applicants to participate in one of them many contribution opportunities we have prior to applying.

Reviewer: Typo?
```suggestion
1. Learn about our [contribution process](https://developers.mattermost.com/contribute/getting-started/). We encourage applicants to participate in one of the many contribution opportunities we have prior to applying.
```

> 1. Review our idea list below for any that might that interest you, all of the details are listed in GitHub Issues on our Community repository.

Reviewer: Typo? Also, maybe we can link at the whole list here
```suggestion
1. Review our idea list below for any that might interest you, all of the details are listed [in GitHub Issues on our Community repository](https://github.com/mattermost/community/issues?q=is%3Aissue+is%3Aopen+%22GSoC+Idea%3A%22).
```

1. Communicate! 
   * Sign up for an account on our [community collaboration platform](https://community.mattermost.com). Join the [GSoC channel](https://community.mattermost.com/core/channels/google-summer-of-code) to chat with mentors and other contributors.
   * Leave any questions you have about the projects as comments on the GitHub issues linked below.

>    * If you have general questions about participating in GSoC with Mattermost, send an email to gsoc@mattermost.com

Reviewer: Should we direct them to the channel first?
```suggestion
   * If you have general questions about participating in GSoC with Mattermost, ask in the [GSoC channel](https://community.mattermost.com/core/channels/google-summer-of-code) or send an email to gsoc@mattermost.com
```

> 1. Work with your mentors on your application, all applications must go through Google's official application process.
> 1. Sumit your application before the deadline.

Reviewer: These read a bit weird. What about this?
```suggestion
1. Work with the project mentors on your application: they are there to help you write a good proposal!
1. Submit your application before the deadline. Note that all applications must go through Google's official application process.
```

Make sure to take a look at [the full Google Summer of Code timeline](https://summerofcode.withgoogle.com/how-it-works/#timeline) for more details!

## Mentors
Here are our mentors for this year, the links lead to their profiles on our community server

-   Technical Engineering Mentors:
    -   Devin Binnie ([@devin.binnie](https://community.mattermost.com/core/messages/@devin.binnie))
    -   Daniel Espino García ([@daniel.espino.garcia](https://community.mattermost.com/core/messages/@daniel.espino.garcia))
    -   Alejandro García Montoro ([@alejandro.garcia](https://community.mattermost.com/core/messages/@alejandro.garcia))

>     -   Elias N ([@elias.nahum](https://community.mattermost.com/core/messages/@elias.nahum))

Reviewer: Poor Elias does not have his last name!
```suggestion
    -   Elias Nahum ([@elias.nahum](https://community.mattermost.com/core/messages/@elias.nahum))
```

    -    Claudio Costa ([@claudio.costa](https://community.mattermost.com/core/messages/@claudio.costa))
-   Product mentors:
    -   Eric Sethna ([@eric.sethna](https://community.mattermost.com/core/messages/@eric.sethna))
    -   Asaad Mahmood ([@asaad.mahmood](https://community.mattermost.com/core/messages/@asaad.mahmood))
-   Organizational Mentors:
    -   Ben Lloyd Pearson([@ben.pearson](https://community.mattermost.com/core/messages/@ben.pearson))

## Ideas list

All project ideas are listed as issues on our community GitHub repo. Here are the summaries:
* [Playbooks mobile app](https://github.com/mattermost/community/issues/171) - Build a proof of concept Playbooks integration for the Mattermost mobile app.
* [Boards mobile app](https://github.com/mattermost/community/issues/170) - Build a proof of concept Boards integration for the Mattermost mobile app.
* [CI pipeline analysis](https://github.com/mattermost/community/issues/169) - Publish a plugin to analyze the CI pipelines in Mattermost to alert the team of build and test failures

> * [Speech to Text Integration](https://github.com/mattermost/community/issues/168) - Publish a plugin to analyze the CI pipelines in Mattermost to alert the team of build and test failures

Reviewer: The descriptions of these are duplicated.

> * [Post Edit History](https://github.com/mattermost/community/issues/167) - Publish a plugin to view the history of edits in posts and improve message history auditing

Reviewer: ```suggestion
* [Post Edit History](https://github.com/mattermost/community/issues/167) - Publish a plugin to view the history of edits in posts and improve message history auditing.
```

* [Recordings for Calls](https://github.com/mattermost/community/issues/166) - Design and implement a recordings service for Calls.
* [Qualify posts importance](https://github.com/mattermost/community/issues/165) - Design and implement a system that introduces the concept of importance to Mattermost messages.
* [Post timed reminders](https://github.com/mattermost/community/issues/164) - Publish a plugin that manages timed reminders on posts that should be read later.
* [Username aliases](https://github.com/mattermost/community/issues/163) - Design and implement a system to manage username aliases.

> * [Custom keyboard shortcuts](https://github.com/mattermost/community/issues/162) - Design and implement a system to let users customize their keyboard shortcuts

Reviewer: ```suggestion
* [Custom keyboard shortcuts](https://github.com/mattermost/community/issues/162) - Design and implement a system to let users customize their keyboard shortcuts.
```

* [Community moderation plugin](https://github.com/mattermost/community/issues/160) - Provide the tools a community needs to moderate activity on their Mattermost instance.
* [Coffee time plugin](https://github.com/mattermost/community/issues/161) - Publish a plugin that helps Mattermost users connect for 1:1 casual conversations and improve remote community building.

### Submit Your Own Idea
If none of the ideas in this list speak to you, you can always propose new ones! Maybe you’ve been wanting to build a specific plugin for a while, you have an amazing core feature in mind that would bring the product to a whole new level, or you want to deep dive into the performance of the server to further optimize it. 

> Ideas proposed by students are more than welcome, don’t hesitate to send them our way! As with all the other projects, you need to take into account that the proposals would need to fit in the overall philosophy of the project: Our product managers will take a look before we can approve them, and we can always discuss the details with the whole team if we decide to move forward with it.

Reviewer: This year they might not be students :)
```suggestion
Ideas proposed by contributors are more than welcome, don’t hesitate to send them our way! As with all the other projects, you need to take into account that the proposals would need to fit in the overall philosophy of the project: Our product managers will take a look before we can approve them, and we can always discuss the details with the whole team if we decide to move forward with it.
```

> If you want to propose your own idea, share your idea as early as possible in our [GSoC channel](https://community.mattermost.com/core/channels/google-summer-of-code
> ).

Reviewer: Formatting:
```suggestion
If you want to propose your own idea, share your idea as early as possible in our [GSoC channel](https://community.mattermost.com/core/channels/google-summer-of-code).
```
